src/runtime/virtcontainers/nydusd.go

```diff
@@ -0,0 +1,424 @@
+	if err != nil {
+		nd.Logger().WithError(err).WithField("path", nd.sockPath).Warn("removing nydusd socket failed")
+	}
+	err = os.Remove(nd.apiSockPath)
+	if err != nil {
+		nd.Logger().WithError(err).WithField("path", nd.apiSockPath).Warn("removing nydusd api socket failed")
+	}
+	return nil
+}
+
+func (nd *nydusd) Logger() *logrus.Entry {
+	return hvLogger.WithField("subsystem", "nydusd")
+}
+
+func (nd *nydusd) kill(ctx context.Context) (err error) {
+	span, _ := katatrace.Trace(ctx, nd.Logger(), "kill", nydusdTracingTags)
+	defer span.End()
+
+	if nd.pid <= 0 {
+		nd.Logger().WithField("invalid-nydusd-pid", nd.pid).Warn("cannot kill nydusd")
+		return nil
+	}
+	err = syscall.Kill(nd.pid, syscall.SIGTERM)
+	if err != nil {
+		nd.Logger().WithError(err).Warn("kill nydusd err")
+	}
+	nd.pid = 0
+	return err
+}
+
+type BuildTimeInfo struct {
+	PackageVer string `json:"package_ver"`
+	GitCommit  string `json:"git_commit"`
+	BuildTime  string `json:"build_time"`
+	Profile    string `json:"profile"`
+	Rustc      string `json:"rustc"`
+}
+
+type DaemonInfo struct {
+	ID      string        `json:"id"`
+	Version BuildTimeInfo `json:"version"`
+	State   string        `json:"state"`
+}
+
+type ErrorMessage struct {
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+type MountRequest struct {
+	FsType string `json:"fs_type"`
+	Source string `json:"source"`
+	Config string `json:"config"`
+}
+
+func NewMountRequest(fsType, source, config string) *MountRequest {
+	return &MountRequest{
+		FsType: fsType,
+		Source: source,
+		Config: config,
+	}
+}
+
+type Interface interface {
+	CheckStatus() (DaemonInfo, error)
+	Mount(string, *MountRequest) error
+	Umount(sharedMountPoint string) error
+}
+
+type NydusClient struct {
+	httpClient *http.Client
+}
+
+func NewNydusClient(sock string) (Interface, error) {
+	transport, err := buildTransport(sock)
+	if err != nil {
+		return nil, err
+	}
+	return &NydusClient{
+		httpClient: &http.Client{
+			Timeout:   defaultHttpClientTimeout,
+			Transport: transport,
+		},
+	}, nil
+}
+
+func waitUntilSocketReady(sock string) error {
+	return retry.Do(func() error {
+		if _, err := os.Stat(sock); err != nil {
+			return err
+		}
+		return nil
+	},
+		retry.Attempts(3),
+		retry.LastErrorOnly(true),
+		retry.Delay(100*time.Millisecond))
+}
+
+func buildTransport(sock string) (http.RoundTripper, error) {
+	err := waitUntilSocketReady(sock)
+	if err != nil {
+		return nil, err
+	}
+	return &http.Transport{
+		MaxIdleConns:          maxIdleConns,
+		IdleConnTimeout:       idleConnTimeout,
+		ExpectContinueTimeout: expectContinueTimeout,
+		DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+			dialer := &net.Dialer{
+				Timeout:   dialTimout,
+				KeepAlive: keepAlive,
+			}
+			return dialer.DialContext(ctx, "unix", sock)
+		},
+	}, nil
+}
+
+func (c *NydusClient) CheckStatus() (DaemonInfo, error) {
+	resp, err := c.httpClient.Get(fmt.Sprintf("http://unix%s", infoEndpoint))
```

Comment:
This path can be calculated at compile time so to avoid hard-codeing `http://unix` here why not change the `const` to include them? For example:

```go
const infoEndpoint = "http://unix/api/v1/daemon"
```

Similar comment for the `Mount()` method below.

Reply:
done